### ISSUE TITLE
[Issue]: Unable to identify TV show

### ISSUE THREAD

**Issue by y2kboy23** [jellyfin/jellyfin#11479]
### Please describe your bug

Jellyfin mis-identified 1923 TV Show in my library.  When I go through the identify process, I select the correct entry but the metadata never downloads.

### Reproduction Steps

Select Identify..
Choose the correct item after searching
Click OK
Item is never updates

### Jellyfin Version

Unstable (master branch)

### Environment

```markdown
Ubuntu 20.04 with Jellyfin running in Docker.
```

### Jellyfin logs

```shell
The only line in the logs about this is: 
[2024-05-02 04:06:49.765 +00:00] [INF] [62] MediaBrowser.Providers.TV.SeriesMetadataService: Creating Season "Season Unknown" entry for "1923"
[2024-05-02 04:12:56.726 +00:00] [INF] [46] MediaBrowser.Providers.TV.SeriesMetadataService: Creating Season "Season Unknown" entry for "Archer"
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by crobibero** [comment]
What is your exact naming scheme for this show?

**Comment by y2kboy23** [comment]
They're in folders.  Show/Season/Episode files.

The files themselves contains 1923.S01E01 as an example.

The folder and file names were not an issue with 10.8.13.

**Comment by y2kboy23** [comment]
I attempted to start without a library and let the scan add the shows to the library.  In reviewing the logs, I'm now seeing this: 

```
[2024-05-02 17:42:48.012 +00:00] [INF] [43] MediaBrowser.Providers.TV.SeriesMetadataService: Removing virtual season null in series "Archer"
[2024-05-02 17:42:48.012 +00:00] [INF] [43] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Season", Name: "Season Unknown", Path: "", Id: a076b25f-b8b4-f3e8-345f-4267f2ec23e8
```

My logs have similar entries for more of the shows than what I originally referenced.  Also, 1923 has not been added to my library but that only has one season.

**Comment by jorensanbar** [comment]
Use this name in folders:

1923 (2022) [tmdbid=157744]  / Season 01 /  1923 - S01E01.mkv

**Comment by devenbj** [comment]
I'm seeing similar behavior in 10.9.0 in Movies. For example, two movies with the same name (different years). My current automation does not include adding the year to the movie (I expect to need to correct these)... one of them will append a "1" to the end of the filename (on my system). For example, "Aladdin" and "Aladdin 1". The one with the filename "Aladdin 1" is wrong. After I choose identify and pick the correct movie, it does not change. If I rename the file with the year and re-scan, then it picks the correct metadata. 

I can manually change the name of the files... but this behavior is not expected. Since I have the option to manually Identify a Movie or TV Show in the GUI without renaming the file, I believe the one I select should be used. That is not happening for me. The logs show no errors. They claim to successfully match to the incorrect metadata. The actual movie it is supposed to match is the 2019 release of the movie.

`[2024-05-12 00:00:35.568 -05:00] [INF] Setting provider id's to item 34727411-c354-6e52-a6ac-84d45b7a18b8-"Aladdin and Ali Baba: Stories from the 1001 Nights?": [("Tmdb": "420817")]`

The behavior changed with the upgrade to v 10.9.0. I have restarted the server multiple times to ensure all plugin updates have completed. I have also force reloaded to clear the browser cache. I also tried on a different browser. I am using Brave and Chrome on Ubuntu. The server is running Ubuntu 22.04.

It appears to me that the metadata fetching is going off of the filename, even when a specific item is selected in the Identify gui. This happens on other movies as well.
